azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/collection/ResourcesInGroup.java

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.resources.fluentcore.collection;
+
+public interface ResourcesInGroup<T, DefinitionT> extends
+    SupportsListing<T>,
+    SupportsCreating<DefinitionT>,
+    SupportsDeleting {}
```

Comment:
actually, it's always by ID so far. It wouldn't work otherwise without the group name. Which we don't know when you say:
azure.publicIpAddresses().get(...).

Note we have both:
```java
.getByGroup(groupName, name)
```
what i'm saying is that for the user's convenience, when you're in InGroup, you'd just want `.getByName(resourceName)` rather than have to also provide the group name - which is already implicitly known - or the id, which you'd have to generate.